lab-dustin/lib/fp.js

```diff
@@ -0,0 +1,27 @@
+exports.mapFunction = function(array, callback){
+  var newArray = Array.prototype.map.call(array, item => item * 2);
+  callback(newArray);
+  console.log(newArray);
+};
+
+exports.reduceFunction = function(array, callback){
+  var reduceVal = Array.prototype.reduce.call(array, (acc,item) => acc += item);
```

Comment:
This should accept a function to replace the built-in `(acc, item) => acc += item`.

Also, note that `acc += item` here has the same impact as `acc + item`; the value assigned to `acc` doesn't matter; it's the value returned from that function that is used as the next `acc`.